### PR TITLE
Add a proxy to liquid esplora api

### PR DESCRIPTION
**Pull request by yaslama**
WIP.
We need to add the right checks

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -370,11 +374,20 @@ func main() {
...
+
+	proxy := httputil.NewSingleHostReverseProxy(liquidEsploraBaseURL)
 	mux := http.NewServeMux()
 	mux.HandleFunc("/fees/v1/btc-fee-estimates.json", func(w http.ResponseWriter, req *http.Request) {
 		w.Header().Set("Content-Type", "application/json")
 		w.Write([]byte(feeEstimates))
 	})
+	mux.HandleFunc(fmt.Sprint("POST ", liquidAPIPrefix, "/tx"), liquid.BroadcastHandler(liquidAPIPrefix, proxy, liquidEsploraBaseURL))
+	mux.HandleFunc(fmt.Sprint("GET ", liquidAPIPrefix, "/mempool"), liquid.MempoolHandler(liquidAPIPrefix, proxy, liquidEsploraBaseURL))
+	mux.HandleFunc(fmt.Sprint("GET ", liquidAPIPrefix, "/mempool/"), liquid.MempoolHandler(liquidAPIPrefix, proxy, liquidEsploraBaseURL))
```
Comment (roeierez) [inline server.go:390]: Here we need the transaction of a specific script hash: https://github.com/blockstream/esplora/blob/master/API.md#get-scripthashhashtxs
e.g:
GET /scripthash/:hash/txs

File: liquid/liquid.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+		return errors.New("bad swap id")
+	}
+	s := struct {
+		Status string `json:"status"`
+	}{}
+	err = json.NewDecoder(r.Body).Decode(&s)
+	if err != nil {
+		return err
+	}
+	if s.Status != "swap.created" {
```
Comment (dangeross) [inline liquid/liquid.go:36]: I think the status here could also be "invoice.set" for submarine swaps

Comment (roeierez) [reply liquid/liquid.go:36]: Yes, @yaslama Ross is right. invoice.set means the swapper has verified the payment info and ready for the lockup.

File: liquid/liquid.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	"net/http/httputil"
+	"net/url"
+	"time"
+
+	"github.com/VictoriaMetrics/fastcache"
+	"github.com/breez/server/breez"
+)
+
+func checkSwapId(fc *fastcache.Cache, apiURL, swapID string, body []byte) error {
+	c := &http.Client{Timeout: 2 * time.Second}
```
Comment (roeierez) [inline liquid/liquid.go:20]: Seems too short which might cause broadcasts to fail. I would go with 10 seconds.

File: liquid/liquid.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+		Status string `json:"status"`
+	}{}
+	err = json.NewDecoder(r.Body).Decode(&s)
+	if err != nil {
+		return err
+	}
+	if s.Status != "swap.created" {
+		return errors.New("bad status")
+	}
+	h := sha256.Sum256(body)
```
Comment (roeierez) [inline liquid/liquid.go:39]: Do we need to compare the body? Why not only check if swap_id exists? We are only going to use it for this endpoint.

Comment (yaslama) [reply liquid/liquid.go:39]: Because if the tx was not broadcasted, we want to be able to retry to broadcast the same tx

**Review by roeierez — APPROVED** [review]

**Review by dangeross — APPROVED** [review]

**Pull request merged**